MAINTAINERS.yml updates

nashif opened the pull request:
The changes here are the result of running all open pull-requests through a script to identify the assignee and finding many gaps or old data. Some of the changes here are coming from pending PRs that were waiting for some TSC confirmation, others are done based on pending maintainer and collaborator changes that were not submitting yet.

Two noteable changes:

- CMake/Build system: This is now limited to just the infrastructure and not every single Cmake file in the tree.
- Documentation: ditto, this is just for the doc infrastructure and not for every RST file in the tree.

The reason for the above two changes is the fact that most PRs would have some cmake or doc changes and assignees in this case should not be the owners of the build system or the doc infra, instead we should assign to the subsystem being changed. (Otherwise majority of PRs would be assigned to the doc and build system maintainers).

nashif committed:
MAINTAINERS: MAINTAINERS.yml updates

The changes here are the result of running all open pull-requests
through a script to identify the assignee and findi ng many gaps or old
data. Some of the changes here are coming from pending PRs that were
waiting for some TSC confirm ation, others are done based on pending
maintainer and collaborator changes that were not submitting yet.

Two noteable changes:

- CMake/Build system: This is now limited to just the infrastructure and
not every single Cmake file in the tree.  - Documentation: ditto, this
is just for the doc infrastructure and not for every RST file in the
tree.

The reason for the above two changes is the fact that most PRs would
have some cmake or doc changes and assignees in this case should not be
the owners of the build system or the doc infra, instead we should
assign to the subsystem being changed. (Otherwise majority of PRs would
be assigned to the doc and build system maintainers).

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

ioannisg commented on MAINTAINERS.yml:
```diff
@@ -1093,7 +1149,92 @@ SPARC arch:
+    labels:
+        - "platform: X86"
+
+Intel Platformas (Xtensa):
```
Platforms*

ioannisg reviewed (changes requested):
Just a typo.
Otherwise, this looks like a good enhancement.

henrikbrixandersen commented on MAINTAINERS.yml:
```diff
@@ -439,7 +461,13 @@ Documentation:
-    status: orphaned
+    status: maintained
+    maintainers:
+      - albertofloyd
+    collaborators:
+      - VenkatKotakonda
+      - scottwcpg
+      - albertofloyd
```
This lists @albertofloyd as both maintainer and collaborator?

nashif replied:
no harm I guess, a maintainer is still a collab

henrikbrixandersen commented on MAINTAINERS.yml:
```diff
@@ -1093,7 +1149,92 @@ SPARC arch:
+  maintainers:
+    - MaureenHelm
+  files:
+    - boards/arm/mimx*/
+    - boards/arm/frdm_k*/
+    - boards/arm/lpcxpress*/
```
We could add `boards/arm/twr_*` here as well.

nashif committed:
MAINTAINERS: replace CODEOWNERS

The Maintairs file will be used to replace the code owners file and will
be used to add reviewers (collaborators + maintainers of affected
areas).

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
MAINTAINERS: update guidelines

Update guidelines regarding updates to maintainers in the MAINTAINER
file.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

mbolivar-nordic commented on MAINTAINERS.yml:
```diff
@@ -1095,7 +1173,93 @@ SPARC arch:
+    maintainers:
+        - ioannisg
+    files:
+        - boards/arm/nrf*/
```
I think `boards/arm/*nrf*/` would be better, to catch things like `thingy52_nrf52832`.

mbolivar-nordic commented on doc/contribute/project_roles.rst:
```diff
@@ -194,7 +194,17 @@ Generic guidelines for deciding and filling in the Maintainers' list
+  directories to the tree.
+* Major changes to the file, including the addition of new areas with new maintainers
+  should come in as standalone pull-requests.
+* Depending on the change, review by the TSC will be required. If additinal
```
`additional`

ioannisg commented on doc/contribute/project_roles.rst:
```diff
@@ -196,7 +194,17 @@ Generic guidelines for deciding and filling in the Maintainers' list
+  in a standalone commit alongside other changes introducing new files and
+  directories to the tree.
+* Major changes to the file, including the addition of new areas with new maintainers
+  should come in as standalone pull-requests and require TSC review.
```
changes in maintainership
